### PR TITLE
feat(s3): add path prefix support for S3 storage backups

### PR DESCRIPTION
**Pull request by karlspace**
## Changes
- Add optional "Path Prefix" field to S3 storage configuration
- Add `path` column to `s3_storages` table via migration
- Apply path prefix when uploading backups to S3 in `DatabaseBackupJob`
- Apply path prefix when deleting backups from S3 in `deleteBackupsS3()` helper
- Add unit tests for path attribute normalization
- Allows storing backups from multiple Coolify instances in a single S3 bucket

### Modified Files:
- `database/migrations/2025_12_26_000001_add_path_to_s3_storages_table.php` - New migration
- `app/Livewire/Storage/Form.php` - Add path property, validation, and sync
- `resources/views/livewire/storage/form.blade.php` - Add path input field with helper text
- `app/Jobs/DatabaseBackupJob.php` - Apply path prefix when uploading to S3
- `bootstrap/helpers/databases.php` - Apply path prefix when deleting from S3
- `tests/Unit/S3StorageTest.php` - Add tests for path normalization

### Use Case:
This feature enables users to configure a path prefix for S3 backups, making it possible to store backups from multiple Coolify instances in a single S3 bucket:
- Instance 1: `s3://bucket/production/data/coolify/backups/...`
- Instance 2: `s3://bucket/staging/data/coolify/backups/...`

### Test Results:
docker exec coolify ./vendor/bin/pest tests/Unit/S3StorageTest.php

   PASS  Tests\Unit\S3StorageTest
  ✓ S3Storage model has correct cast definitions                         0.12s
  ✓ S3Storage isUsable method returns is_usable attribute value          0.01s
  ✓ S3Storage awsUrl method constructs correct URL format
  ✓ S3Storage model is guarded correctly
  ✓ S3Storage path attribute normalizes path correctly
  ✓ S3Storage path attribute handles various path formats

  Tests:    6 passed (17 assertions)
  Duration: 1.95s

docker exec coolify ./vendor/bin/pint --test

  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................
  ............................................................................

  ──────────────────────────────────────────────────────────────────── Laravel
    PASS   ........................................................ 1140 files

s3_bucket_path_settings

### CODE REVIEW

File: tests/Unit/S3StorageTest.php [hunk 1]
```diff
@@ -51,3 +51,43 @@
...
+    expect($s3Storage->path)->toBe('/instance-1');
+
+    // Nested path
+    $s3Storage->path = 'production/backups/db';
+    expect($s3Storage->path)->toBe('/production/backups/db');
+
+    // Path with special characters
+    $s3Storage->path = 'my-instance_2024.backups';
+    expect($s3Storage->path)->toBe('/my-instance_2024.backups');
+});
```
[the remark this answers is not in the document]
Comment (karlspace) [reply tests/Unit/S3StorageTest.php]: Addressed in commit - added edge case tests and not_regex:/\.\./ validation to block directory traversal.

**Comment by brunoalbim** [comment]
UP.
It would be great to have the option of a prefix in the S3 path.
